api/src/indicators.py

```diff
@@ -486,10 +478,22 @@ def patch_annotation(payload: MetadataSchema.MetaModel, indicator_id: str):
 
         body = json.loads(payload.json(exclude_unset=True))
 
+        # Handles datasets being regsitered with no date.
+        if body.get("annotations"):
+            if not body.get("annotations").get("date", []):
```

Comment:
Actually, I just realized- the UI warns the user that a missing `primary` date will default to today's date, but what should we do if the user did annotate date field, but is not marked a primary? I don't think this code handles that scenario.